File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
def solution(word):
    answer = 0
    
    def find_word(string):
        nonlocal answer
        
        if string == word: return True
        if len(string) == 5: return False
        
        for char in ['A', 'E', 'I', 'O', 'U']:
            answer += 1
            if find_word(string + char): return True
    
    find_word("")
    
    return answer

File: 프로그래머스/2/84512. 모음 사전/모음 사전.js
function solution(word) {
    let answer = 0;
    
    const find_word = (string) => {
        if (string === word) return true;
        if (string.length === 5) return false;
        
        for (const char of ['A', 'E', 'I', 'O', 'U']) {
            answer++;
            if (find_word(string + char)) return true;
        }
    };
    
    find_word("");
    
    return answer;
}